avi/resource_avi_cloud.go

```diff
@@ -236,10 +243,101 @@ func ResourceAviCloudRead(d *schema.ResourceData, meta interface{}) error {
 	return err
 }
 
+// Verify cloudState is ready to configure mgmt IP
+func isCloudReady(cloudState string) bool {
+	switch cloudState {
+	case
+		"VCENTER_DISCOVERY_COMPLETE_NO_MGMT_NW",
+		"VCENTER_DISCOVERY_COMPLETE",
+		"VCENTER_DISCOVERY_WAITING_DC",
+		"VCENTER_DISCOVERY_ONGOING":
+		return true
+	}
+	return false
+}
+
+// Wait untill cloud is ready for the placement if isCloudInventory is set otherwise will wait untill vcenter
+//inventory gets completed to configure mgmt IP
+func waitForCloudState(cloudUuid string, client *clients.AviClient, maxRetry int, isCloudInventory bool) error {
+	var robj interface{}
+	var err error
+	var inventoryState string
+	var path string
+	if isCloudInventory {
+		path = "api/cloud-inventory?uuid=" + cloudUuid
```

Comment:
we should split this into two functions... no need to try to re-use the retry path.